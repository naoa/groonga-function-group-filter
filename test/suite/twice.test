plugin_register functions/group_filter
table_create applicants TABLE_HASH_KEY ShortText
table_create ipcs TABLE_HASH_KEY ShortText
table_create docs TABLE_HASH_KEY ShortText
column_create docs applicants COLUMN_VECTOR applicants
column_create docs ipcs COLUMN_VECTOR ShortText
load --table docs
[
{"_key": "JP2001213456", "applicants": ["三洋", "三洋オプテックデザイン"], "ipcs": ["G06F", "G04A"]},
{"_key": "JP2001213457", "applicants": ["パナソニック"], "ipcs": ["G04A"]},
{"_key": "JP2001213458", "applicants": ["ソニー"], "ipcs": ["G06F", "G04B"]},
{"_key": "JP2001213459", "applicants": ["パナソニック", "三洋"], "ipcs": ["G06F"]}
]
column_create applicants applicants COLUMN_INDEX docs applicants
column_create ipcs ipcs COLUMN_INDEX docs ipcs
select docs \
  --filter '_id >= 1 && group_filter("applicants", 2) && group_filter("ipcs", 2)' \
  --output_columns _id,_score,_key,applicants,ipcs \
  --drilldowns[applicants_ipcs].keys applicants,ipcs \
  --drilldowns[applicants_ipcs].columns[applicant].stage initial \
  --drilldowns[applicants_ipcs].columns[applicant].type applicants \
  --drilldowns[applicants_ipcs].columns[applicant].flags COLUMN_SCALAR \
  --drilldowns[applicants_ipcs].columns[applicant].value '_key[0]' \
  --drilldowns[applicants_ipcs].columns[ipc].stage initial \
  --drilldowns[applicants_ipcs].columns[ipc].type ipcs \
  --drilldowns[applicants_ipcs].columns[ipc].flags COLUMN_SCALAR \
  --drilldowns[applicants_ipcs].columns[ipc].value '_key[1]' \
  --drilldowns[applicants_ipcs].output_columns applicant,ipc,_nsubrecs \
  --drilldowns[applicants_ipcs].sort_keys applicant,ipc \
  --drilldowns[applicants_ipcs].limit -1 \
  --drilldowns[top_applicants].table applicants_ipcs \
  --drilldowns[top_applicants].keys applicant \
  --drilldowns[top_applicants].calc_types SUM \
  --drilldowns[top_applicants].calc_target _nsubrecs \
  --drilldowns[top_applicants].output_columns _key,_sum \
  --drilldowns[top_applicants].sort_keys -_sum \
  --drilldowns[top_applicants].limit 2 \
  --drilldowns[top_ipcs].table applicants_ipcs \
  --drilldowns[top_ipcs].keys ipc \
  --drilldowns[top_ipcs].calc_types SUM \
  --drilldowns[top_ipcs].calc_target _nsubrecs \
  --drilldowns[top_ipcs].output_columns _key,_sum \
  --drilldowns[top_ipcs].sort_keys -_sum \
  --drilldowns[top_ipcs].limit 2
