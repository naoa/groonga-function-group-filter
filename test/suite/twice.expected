plugin_register functions/group_filter
[[0,0.0,0.0],true]
table_create applicants TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create ipcs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create docs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create docs applicants COLUMN_VECTOR applicants
[[0,0.0,0.0],true]
column_create docs ipcs COLUMN_VECTOR ShortText
[[0,0.0,0.0],true]
load --table docs
[
{"_key": "JP2001213456", "applicants": ["三洋", "三洋オプテックデザイン"], "ipcs": ["G06F", "G04A"]},
{"_key": "JP2001213457", "applicants": ["パナソニック"], "ipcs": ["G04A"]},
{"_key": "JP2001213458", "applicants": ["ソニー"], "ipcs": ["G06F", "G04B"]},
{"_key": "JP2001213459", "applicants": ["パナソニック", "三洋"], "ipcs": ["G06F"]}
]
[[0,0.0,0.0],4]
column_create applicants applicants COLUMN_INDEX docs applicants
[[0,0.0,0.0],true]
column_create ipcs ipcs COLUMN_INDEX docs ipcs
[[0,0.0,0.0],true]
select docs   --filter '_id >= 1 && group_filter("applicants", 2) && group_filter("ipcs", 2)'   --output_columns _id,_score,_key,applicants,#group_applicants,ipcs,#group_ipcs   --drilldowns[applicants_ipcs].keys #group_applicants,#group_ipcs   --drilldowns[applicants_ipcs].columns[applicant].stage initial   --drilldowns[applicants_ipcs].columns[applicant].type applicants   --drilldowns[applicants_ipcs].columns[applicant].flags COLUMN_SCALAR   --drilldowns[applicants_ipcs].columns[applicant].value '_key[0]'   --drilldowns[applicants_ipcs].columns[ipc].stage initial   --drilldowns[applicants_ipcs].columns[ipc].type ipcs   --drilldowns[applicants_ipcs].columns[ipc].flags COLUMN_SCALAR   --drilldowns[applicants_ipcs].columns[ipc].value '_key[1]'   --drilldowns[applicants_ipcs].output_columns applicant,ipc,_nsubrecs   --drilldowns[applicants_ipcs].sort_keys applicant,ipc   --drilldowns[applicants_ipcs].limit -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        3
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_score",
          "Int32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "applicants",
          "applicants"
        ],
        [
          "#group_applicants",
          "applicants"
        ],
        [
          "ipcs",
          "ShortText"
        ],
        [
          "#group_ipcs",
          "ShortText"
        ]
      ],
      [
        1,
        4,
        "JP2001213456",
        [
          "三洋",
          "三洋オプテックデザイン"
        ],
        [
          "三洋"
        ],
        [
          "G06F",
          "G04A"
        ],
        [
          "G06F",
          "G04A"
        ]
      ],
      [
        2,
        3,
        "JP2001213457",
        [
          "パナソニック"
        ],
        [
          "パナソニック"
        ],
        [
          "G04A"
        ],
        [
          "G04A"
        ]
      ],
      [
        4,
        4,
        "JP2001213459",
        [
          "パナソニック",
          "三洋"
        ],
        [
          "パナソニック",
          "三洋"
        ],
        [
          "G06F"
        ],
        [
          "G06F"
        ]
      ]
    ],
    {
      "applicants_ipcs": [
        [
          4
        ],
        [
          [
            "applicant",
            "applicants"
          ],
          [
            "ipc",
            "ipcs"
          ],
          [
            "_nsubrecs",
            "Int32"
          ]
        ],
        [
          "三洋",
          "G06F",
          2
        ],
        [
          "三洋",
          "G04A",
          1
        ],
        [
          "パナソニック",
          "G06F",
          1
        ],
        [
          "パナソニック",
          "G04A",
          1
        ]
      ]
    }
  ]
]
